sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/ShuffleExchange.scala

```diff
@@ -47,7 +47,7 @@ case class ShuffleExchange(
     val extraInfo = coordinator match {
       case Some(exchangeCoordinator) =>
         s"(coordinator id: ${System.identityHashCode(exchangeCoordinator)})"
-      case None => ""
+      case _ => ""
```

Comment:
I had left the other match conditions unchanged so that this change itself can be small and contained, because I'd need to verify whether or not the other match conditions should only be invoked with some valid `Option[ExchangeCoordinator]` value.

Anyway, I checked the two other `coordinator match { ... }` cases below, and both of them should still work fine to change to match on `_` instead of `None`. But semantically both of these cases should only be invoked on the Driver side, where the `coordinator` shouldn't be `null`. I'm okay to change these two cases if reviewers suggest so.